includes/class-sensei-guest-user.php

```diff
@@ -59,20 +59,36 @@ public function sensei_create_guest_user_and_login_for_open_course() {
 		) {
 			$user_id = $this->create_guest_student_user();
 			$this->login_user( $user_id );
+			$this->recreate_nonces();
 		}
 	}
 
 	/**
 	 * Check if the course is open access.
 	 *
-	 * @param int $course_id ID of the course.
+	 * @param  int $course_id ID of the course.
 	 * @since  $$next-version$$
 	 * @return boolean|mixed
 	 */
 	private function is_course_open_access( $course_id ) {
 		return get_post_meta( $course_id, 'open_access', true );
 	}
 
+	/**
+	 * Recreate nonce after logging in user invalidates existing one.
+	 *
+	 * @since $$next-version$$
+	 */
+	private function recreate_nonces() {
+		$nonce_field = wp_nonce_field( 'woothemes_sensei_start_course_noonce', 'woothemes_sensei_start_course_noonce', false, false );
+		$pattern     = '/value="([^"]*)"/';
```

Comment:
There is a `wp_create_nonce` function that returns just the token, that should work better here.

Reply:
Oh! Thanks! Updated [here](https://github.com/Automattic/sensei/pull/6269/commits/3ade221bb37d3e01095278a273b3746a0cd91232)